src/controller/CHIPDeviceController.h

```diff
@@ -48,6 +48,38 @@ typedef void (*ErrorHandler)(ChipDeviceController * deviceController, void * app
                              const Inet::IPPacketInfo * pktInfo);
 typedef void (*MessageReceiveHandler)(ChipDeviceController * deviceController, void * appReqState, System::PacketBuffer * payload);
 
+class DLL_EXPORT PersistentStorageDelegate
+{
+public:
+    /**
+     * @brief
+     *   Lookup the key and return it's stringified value
+     *
+     * @param[in] key Key to lookup
+     * @return Value or nullptr if not found
```

Comment:
Should probably document who owns the data and how long that pointer can be expected to stay alive.